Kitty Items Documentation update (#264)
onflow/kitty-items: api/README.md

# Kitty Items API
The Kitty Items API is a RESTful API built with [express](https://expressjs.com/) that sends transactions to Flow using the [Flow JS SDK](https://github.com/onflow/fcl-js/tree/master/packages/sdk). It contains endpoints for both the [kitty-items](src/services/kitty-items.ts) & [storefront](src/services/storefront.ts) services to read & write data from both the Flow blockchain as well as a SQL DB.

## Services

> Kitty Items uses main 4 services to function. These services are used to maintain data integrity between the blockchain & our database as well as allow avenues for users to interact with our data through exposed api endpoints. 

Reviewer: ```suggestion
The Kitty Items API contains four main services. These services synchronize data between the blockchain and the SQL database, and also provide API endpoints consumed by the Kitty Items client-side application. 
```

### [Flow Service](src/services/flow.ts)

> This service contains functions that interact with the Flow blockchain using the [onflow/fcl](https://docs.onflow.org/fcl/) package. While it has no exposed endpoints, the methods here used to read, write & authorize data on the chain are used extensively in its sister services & workers.

Reviewer: ```suggestion
This service contains functions that interact with the Flow blockchain using the [FCL JS](https://docs.onflow.org/fcl/) library. While it has no exposed endpoints, its methods are used to read, write and authorize data on the chain are used extensively by its sister services and workers.
```

IMO best to only use ampersands (&) in titles

Author: `While it has no exposed endpoints, its methods ~are~ used to read, write and authorize data on the chain are used extensively by its sister services and workers.`

Notable functions:
- `sendTx()`: Takes a Cadence transaction as an arguement and sends it to write data into the Flow blockchain.
- `executeScript()`: Takes a Cadence script as an arguement and executes it to read data from the Flow blockchain. 
- `getLatestBlockHeight()`: Gets the most recent block from the Flow blockchain.
- `authorizeMinter()`: Returns an asynchronous method to authorize an account to mint Kitty Items.

> ### [Block-cursor Service](src/services/block-cursor.ts)

Reviewer: ```suggestion
### [Block Cursor Service](src/services/block-cursor.ts)
```

> This service contains functions that read & write data to the SQL database to maintain the Kitty Items Marketplace

Reviewer: ```suggestion
This service watches for state-changing events emitted by the Kitty Items contract and updates the SQL database accordingly.
```

Notable functions:
- `findOrCreateLatestBlockCursor()`: Queries the `block_cursor` table for the the last block check by the listing handler worker. If no row is found then insert a record with our current latest block height.
- `updateBlockCursorById()`: Updates the row with the latest block height that has been checked for events.

> ### [Kitty-items Service](src/services/kitty-items.ts)

Reviewer: ```suggestion
### [Kitty Items Service](src/services/kitty-items.ts)
```

Reviewer: Also, I noticed that you took out the sample curl commands. I think those were helpful to show developers how to call the API.

Author: i took out the curl commands since we had flow scripts that sort of abstract away the endpoints but i can see the value in letting them hit the endpoints themselves. ill add the curl commands back in a `Getting Started` section

> This service contains functions that utilize the Flow Service to send/execute transactions/scripts that fetch/manipulate Kitty Items data. These functions are exposed through endpoints to allow users to read & manipulate Kitty Items data on the chain. The Cadence scripts/transactions can also be called in the [Flow-cli](https://docs.onflow.org/flow-cli/) and directly read/write to the Flow blockchain.

Reviewer: ```suggestion
This service contains functions that utilize the Flow Service to send Cadence transactions and scripts that fetch and update Kitty Items data.

You can also run the [transactions and scripts](../cadence) manually using the [Flow CLI](https://docs.onflow.org/flow-cli/).
```

I'd keep this simple.

[Exposed Endpoints](src/routes/kitty-items.ts):
- **POST `/v1/kitty-items/setup`** *[DEPRECATED]*: Sets up account with a resource that stores Kitty Items in the admin minter account. The [startup script](/.ki-scripts/startup.js) directly sends the setup Cadence transaction instead of using this endpoint. Calls [setup_account.cdc](/cadence/transactions/kittyItems/setup_account.cdc)

- **POST `/v1/kitty-items/mint`** *[DEPRECATED]*: Mint a Kitty Item and send it to a recipient account. Calls [mint_kitty_item.cdc](/cadence/transactions/kittyItems/mint_kitty_item.cdc)

> - **POST `/v1/kitty-items/mint-and-list`**: Mint a Kitty Item and send it to a recipient account. Create a listing to sell the Kitty Item immediately after. Called by the admin service account to mint & list items onto the store. Calls [mint_and_list_kitty_item.cdc](/cadence/transactions/kittyItems/mint_and_list_kitty_item.cdc)

Reviewer: ```suggestion
- **POST `/v1/kitty-items/mint-and-list`**: Mint a Kitty Item and immediately list it for sale. Called by the admin account to mint and sell new items. Calls [mint_and_list_kitty_item.cdc](/cadence/transactions/kittyItems/mint_and_list_kitty_item.cdc)
```

I would just say "admin account" instead of "admin service account" because "service account" doesn't mean much outside the context of the emulator.

> - **POST `/v1/kitty-items/transfer`**: Sends a Kitty Item from account to recipient account. Calls [transfer_kitty_item.cdc](/cadence/transactions/kittyItems/transfer_kitty_item.cdc)

Reviewer: ```suggestion
- **POST `/v1/kitty-items/transfer`**: Sends a Kitty Item from the admin account to a recipient account. Calls [transfer_kitty_item.cdc](/cadence/transactions/kittyItems/transfer_kitty_item.cdc)
```
